### PR TITLE
NMS-13386: Add docs to Health-Check Rest API

### PR DESCRIPTION
**Pull request by aramosvizcarra**
Adding Health Rest Api documentation

* JIRA (Issue Tracker): https://issues.opennms.org/browse/NMS-13386

### CODE REVIEW

File: docs/modules/development/pages/rest/health_rest.adoc [hunk 1]
```diff
@@ -0,0 +1,42 @@
+
+= Health Rest Service
+
+Service that allows health rest calls
```
Comment (Bonrob2) [inline docs/modules/development/pages/rest/health_rest.adoc:4]:
```suggestion
Service that allows health rest calls.
```

File: docs/modules/development/pages/rest/health_rest.adoc [hunk 2]
```diff
@@ -0,0 +1,42 @@
...
+
+Currently implemented on Minion and Sentinel by default
```
Comment (Bonrob2) [inline docs/modules/development/pages/rest/health_rest.adoc:6]:
```suggestion
Currently implemented on Minion and Sentinel by default.
```

File: docs/modules/development/pages/rest/health_rest.adoc [hunk 3]
```diff
@@ -0,0 +1,42 @@
...
+
+The following table shows all supported query string parameters and their default values.
+
+[options="header"]
+|===
+| Name              | Default  | Comment
+s| t                 | 5000     | Timeout in milliseconds
+s| maxAgeMs          | 90000    | 
+s| tag               | ''       
+a| Use tag to filter Health Check components. A tag could be: +
```
Comment (Bonrob2) [inline docs/modules/development/pages/rest/health_rest.adoc:32]:
```suggestion
a| Use tag to filter health check components. A tag could be: +
```

File: docs/modules/development/pages/rest/health_rest.adoc [hunk 4]
```diff
@@ -0,0 +1,42 @@
...
+Examples:
+
+* /rest/health
+* /rest/health/probe
+* /rest/health?tag=local
+* /rest/health/probe?tag=local
+
+== GETs (Reading Data)
+
+[options="header, %autowidth"]
```
Comment (Bonrob2) [inline docs/modules/development/pages/rest/health_rest.adoc:17]: Please update table formatting style as described in the "Develop Documentation" section of the docs (https://docs.opennms.com/horizon/29.0.1/development/docs/writing.html#gd-docs-tables). We have recently updated how we do tables.

File: docs/modules/development/pages/rest/health_rest.adoc [hunk 5]
```diff
@@ -0,0 +1,42 @@
...
+|===
+| Resource             | Description
+| `/health`            | Health check results for components. Include health check result details for each element.
+| `/health/prob`       | Health check probe for components. Only indicate if the health is good or not. 
+|===
+
+The following table shows all supported query string parameters and their default values.
+
+[options="header"]
```
Comment (Bonrob2) [inline docs/modules/development/pages/rest/health_rest.adoc:26]: Please update table formatting style as described in the "Develop Documentation" section of the docs (https://docs.opennms.com/horizon/29.0.1/development/docs/writing.html#gd-docs-tables). We have recently updated how we do tables.

**Review by Bonrob2 — CHANGES_REQUESTED** [review]
A few minor changes, but also need to change table syntax to reflect recent changes.

File: docs/modules/development/pages/rest/health_rest.adoc (hunk 5 above)
Comment (aramosvizcarra) [reply docs/modules/development/pages/rest/health_rest.adoc]: Updated base on the documentation

File: docs/modules/development/pages/rest/health_rest.adoc [hunk 6]
```diff
@@ -0,0 +1,43 @@
...
+* /rest/health?tag=local
+* /rest/health/probe?tag=local
+
+== GETs (Reading Data)
+
+[options="header", cols="5,10"]
+|===
+| Resource             | Description
+| `/health`            | Health check results for components. Include health check result details for each element.
+| `/health/prob`       | Health check probe for components. Only indicate if the health is good or not.
```
Comment (cgorantla) [inline docs/modules/development/pages/rest/health_rest.adoc:21]: Shouldn't it be `/health/probe` ?

File: docs/modules/development/pages/rest/health_rest.adoc [hunk 7]
```diff
@@ -0,0 +1,43 @@
...
+ * local + 
+ * rest +
+ * telemetry +
+ * dao +
+ * minion +
+ * elastic +
+ * bundle 
+                                                    | '' 
+|===
+
```
Comment (cgorantla) [inline docs/modules/development/pages/rest/health_rest.adoc:42]: Also add documentation for mixing multiple tags.  

Also define error http codes and when user may get those error codes.

File: docs/modules/development/pages/rest/health_rest.adoc [hunk 8]
```diff
@@ -0,0 +1,43 @@
...
+* /rest/health
+* /rest/health/probe
+* /rest/health?tag=local
+* /rest/health/probe?tag=local
+
+== GETs (Reading Data)
+
+[options="header", cols="5,10"]
+|===
+| Resource             | Description
```
Comment (Bonrob2) [inline docs/modules/development/pages/rest/health_rest.adoc:19]:
```suggestion
| Resource            
| Description
```

File: docs/modules/development/pages/rest/health_rest.adoc [hunk 9]
```diff
@@ -0,0 +1,43 @@
...
+| `/health`            | Health check results for components. Include health check result details for each element.
```
Comment (Bonrob2) [inline docs/modules/development/pages/rest/health_rest.adoc:20]:
```suggestion
| `/health`
| Health check results for components. Include health check result details for each element.
```

File: docs/modules/development/pages/rest/health_rest.adoc (hunk 8 above)
Comment (Bonrob2) [reply docs/modules/development/pages/rest/health_rest.adoc:19]: Table syntax is to have each cell on a separate line, a blank line between rows. (p.s. I hate tables. :-))

File: docs/modules/development/pages/rest/health_rest.adoc [hunk 10]
```diff
@@ -0,0 +1,43 @@
...
+| `/health/prob`       | Health check probe for components. Only indicate if the health is good or not. 
+|===
+
+The following table shows all supported query string parameters and their default values.
+
+[options="header", cols="1,3,1"]
+|===
+| Name               | Description                  | Default
```
Comment (Bonrob2) [inline docs/modules/development/pages/rest/health_rest.adoc:28]:
```suggestion
| Name
| Description
| Default 
```

File: docs/modules/development/pages/rest/health_rest.adoc [hunk 11]
```diff
@@ -0,0 +1,43 @@
...
+| `/health`            | Health check results for components. Include health check result details for each element.
+| `/health/prob`       | Health check probe for components. Only indicate if the health is good or not. 
+|===
+
+The following table shows all supported query string parameters and their default values.
+
+[options="header", cols="1,3,1"]
+|===
+| Name               | Description                  | Default 
+| t                 | Timeout in milliseconds      | 5000
```
Comment (Bonrob2) [inline docs/modules/development/pages/rest/health_rest.adoc:29]:
```suggestion
| t
| Timeout in milliseconds
| 5000 
```

File: docs/modules/development/pages/rest/health_rest.adoc [hunk 12]
```diff
@@ -0,0 +1,43 @@
...
+| `/health/prob`       | Health check probe for components. Only indicate if the health is good or not. 
+|===
+
+The following table shows all supported query string parameters and their default values.
+
+[options="header", cols="1,3,1"]
+|===
+| Name               | Description                  | Default 
+| t                 | Timeout in milliseconds      | 5000 
+| maxAgeMs          | Max age to live miliseconds  | 90000
```
Comment (Bonrob2) [inline docs/modules/development/pages/rest/health_rest.adoc:30]:
```suggestion
| maxAgeMs
| Max age to live milliseconds
| 90000 
```

**Review by Bonrob2 — COMMENTED** [review]
a few more updates to the table syntax.

File: docs/modules/development/pages/rest/health_rest.adoc (hunk 10 above)
Comment (aramosvizcarra) [reply docs/modules/development/pages/rest/health_rest.adoc:30]: OK

File: docs/modules/development/pages/rest/health_rest.adoc (hunk 7 above)
Comment (aramosvizcarra) [reply docs/modules/development/pages/rest/health_rest.adoc:50]: Not sure about the errors. from what I can see there will be only 200 or 404. Are you referring to the Status enum ?

**Review by cgorantla — APPROVED** [review]
LGTM!

File: docs/modules/development/pages/rest/health_rest.adoc [hunk 13]
```diff
@@ -0,0 +1,92 @@
+
+= Health Rest Service
+
+Service that allows health rest calls on all _HealthCheck_ components available .
```
Comment (Bonrob2) [inline docs/modules/development/pages/rest/health_rest.adoc:4]:
```suggestion
Service that allows health rest calls on all available health check components.
```

File: docs/modules/development/pages/rest/health_rest.adoc [hunk 14]
```diff
@@ -0,0 +1,92 @@
...
+| Description
+| Default 
+| t
+| Timeout in milliseconds
+| 5000 
+| maxAgeMs
+| Max age to live milliseconds
+| 90000 
+| tag                     
+| Use tag to filter Health Check components. A tag could be: +
```
Comment (Bonrob2) [inline docs/modules/development/pages/rest/health_rest.adoc:42]:
```suggestion
| Use tag to filter health check components. A tag could be: +
```

File: docs/modules/development/pages/rest/health_rest.adoc (hunk 13 above)
Comment (Bonrob2) [reply docs/modules/development/pages/rest/health_rest.adoc:4]: Sorry, I didn't notice this before, but I checked the main docs, and we tend to use "health check".

File: docs/modules/development/pages/rest/health_rest.adoc [hunk 15]
```diff
@@ -0,0 +1,92 @@
...
+ * broker +
+ * telemetry +
+ * dao +
+ * minion +
+ * elastic +
+ * bundle 
+| '' 
+|===
+
+Requests without parameters will return all HealthCheck components available
```
Comment (Bonrob2) [inline docs/modules/development/pages/rest/health_rest.adoc:54]:
```suggestion
Requests without parameters will return all available health check components. 
```

File: docs/modules/development/pages/rest/health_rest.adoc [hunk 16]
```diff
@@ -0,0 +1,92 @@
...
+            "description": "Verifying Listener JTI-Listener (org.opennms.netmgt.telemetry.listeners.UdpListener)",
+            "status": "Success"
+        },
+        ...
+    ]
+}
+
+----
+
+== Http Error Codes
```
Comment (Bonrob2) [inline docs/modules/development/pages/rest/health_rest.adoc:80]:
```suggestion
== HTTP Error Codes
```

**Review by Bonrob2 — DISMISSED** [review]
Minor updates to the text, but otherwise approved.

**Review by Bonrob2 — APPROVED** [review]
LGTM!

**Comment by cgorantla** [comment]
Missed this earlier, We probably need to add this doc here: https://github.com/OpenNMS/opennms/blob/19993e9829a255ec579f7103834eb1ca15307407/docs/modules/development/nav.adoc

@aramosvizcarra you can open another PR for it

**Pull request merged**